adit/mass_transfer/management/commands/discover_studies.py

```
@@ -0,0 +1,166 @@
+            default="",
+            help="Path to write CSV output. If omitted, writes to stdout.",
+        )
+
+    def handle(self, *args, **options):
+        server_name = options["source"]
+        output_path = options["output"]
+
+        try:
+            server = DicomServer.objects.get(name=server_name)
+        except DicomServer.DoesNotExist:
+            raise CommandError(f"DICOM server '{server_name}' not found.")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)